packages/google_adsense/lib/src/adsense_stub.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'ad_unit_widget.dart';
+
+/// Main class to work with the library
+class Adsense {
+  /// Returns a singleton instance of Adsense library public interface
+  factory Adsense() => _instance ?? Adsense._internal();
+
+  Adsense._internal() {
+    _instance = this;
+  }
+
+  static Adsense? _instance = Adsense._internal();
+
+  /// Initialization API. Should be called ASAP, ideally in the main method of your app.
+  void initialize(String adClient) {
+    throw UnsupportedError('Only supported on web');
+  }
+
+  /// Returns a configurable [AdUnitWidget]
+  AdUnitWidget adUnit(
+      {required String adSlot,
+      String adClient = '',
+      bool isAdTest = false,
+      Map<String, dynamic> adUnitParams = const <String, dynamic>{},
+      String? cssText}) {
+    throw UnsupportedError('Only supported on web');
+  }
+
+  /// Only for use in tests
+  static void resetForTesting() {
+    _instance = null;
+  }
+}
```

Comment:
I think this file should expose a global:

`final AdSense adSense = AdSense();`

So people can use this as:

```dart
  adSense.initialize(...);
  adSense.adUnit(...);
```

Instead of always having to do: `AdSense().things()`

Reply:
I have changed it to be Adsense.instance. Let's discuss if global is a better approach